### PR TITLE
aaron/profile-settings-tax-information

### PR DESCRIPTION
**Pull request by aaron-binary**

### CODE REVIEW

File: src/javascript/app/pages/user/account/settings/personal_details.js [hunk 1]
```diff
@@ -97,7 +108,7 @@ const PersonalDetails = (() => {
         showHideMissingDetails();
     };
 
-    const show_label_if_any_value = ['account_opening_reason', 'citizen', 'place_of_birth'];
+    const show_label_if_any_value = ['account_opening_reason', 'citizen', 'place_of_birth', 'tax_residence', 'tax_identification_number'];
```
Comment (negarn) [inline src/javascript/app/pages/user/account/settings/personal_details.js]: we want to allow them to update it, no?

Comment (negarn) [reply src/javascript/app/pages/user/account/settings/personal_details.js]: just saw the trello comment, nvm

File: src/javascript/app/pages/user/account/settings/personal_details.js [hunk 2]
```diff
@@ -73,6 +73,17 @@ const PersonalDetails = (() => {
...
         }
 
+        if (get_settings.tax_residence) {
+            get_settings.tax_residence =
+                get_settings.tax_residence
+                    .split(',')
+                    .reduce((all_residences, current_residence) =>
+                        `${all_residences + ((residence_list.find(obj => obj.value === current_residence) || {}).text || current_residence)  }, `, ''
+                    )
+                    .slice(0, -2) ||
```
Comment (negarn) [inline src/javascript/app/pages/user/account/settings/personal_details.js]: why not something simple like this?
```
get_settings.tax_residence
    .split(',')
    .map((current_residence) =>
        (residence_list.find(obj => obj.value === current_residence) || {}).text || current_residence
    ).join(', ')
```

Comment (aaron-binary) [reply src/javascript/app/pages/user/account/settings/personal_details.js]: Because I hadn't thought of that approach

**Pull request merged**